yurt/fabfile/setup.py

```diff
@@ -1,73 +1,73 @@
 
 __author__ = 'deanmercado'
 
 YURT_PATH = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
 DJANGO_PROJECT_PATH = os.path.join(YURT_PATH, 'django_project')
 ORCHESTRATION_PROJECT_PATH = os.path.join(YURT_PATH, 'orchestration')
+FABFILE_PATH = os.path.join(YURT_PATH, 'fabfile')
 
 
-def create_settings():
-    fabric_settings = get_fab_settings()
+def create_settings(vault, git_repo):
+    fabric_settings = add_settings(vault, git_repo)
     project_name = get_project_name_from_repo(fabric_settings.get('git_repo'))
     fabric_settings['project_name'] = project_name
     fabric_settings['git_repo_url'] = fabric_settings.get('git_repo')
     fabric_settings['repo_name'] = get_project_name_from_repo(fabric_settings.get('git_repo'), False)
     return fabric_settings, project_name
 
 
-def create_project():
+def create_project(fabric_settings, project_name):
     """
-    Creates Django project by copying over
-    :return: Void
+    Creates Django project by copying over stuff
     """
-    fabric_settings, project_name = create_settings()
-    local("cp -rf {0} ./{1}".format(os.path.join(DJANGO_PROJECT_PATH, "*"), project_name))
+    run("cp -rf {0} ./{1}".format(os.path.join(DJANGO_PROJECT_PATH, "*"), project_name))
     recursive_file_modify(os.path.abspath("./{0}".format(project_name)), fabric_settings)
 
 
-def load_orchestration_and_requirements():
+def load_orchestration_and_requirements(fabric_settings, project_name):
     """
     Copies over the orchestration directory and requirements.txt file to current directory
-    :return: Void
     """
-    fabric_settings, project_name = create_settings()
-    local('cp -rf {0} ./{1}'.format(ORCHESTRATION_PROJECT_PATH, project_name))
-    local('cp -f {0} ./{1}'.format(os.path.join(YURT_PATH, 'requirements.txt'), project_name))
+    run('cp -rf {0} ./{1}'.format(ORCHESTRATION_PROJECT_PATH, project_name))
+    run('cp -f {0} ./{1}'.format(os.path.join(YURT_PATH, 'requirements.txt'), project_name))
+    run('cp -f {0} ./{1}/.gitignore'.format(os.path.join(FABFILE_PATH, 'gitignore.template'), project_name))
     recursive_file_modify('./{0}/orchestration'.format(project_name), fabric_settings)
 
 
-def enable_git_repo():
+def enable_git_repo(fabric_settings, project_name):
     """
     Sets up git repository in project direcory
-    :return: Void
     """
-    fabric_settings, project_name = create_settings()
     if project_name not in os.listdir('.'):
-        local('mkdir {0}'.format(project_name))
+        run('mkdir {0}'.format(project_name))
 
-    with lcd(project_name):
-        with settings(warn_only=True):
-            local('git init')
-            local('git remote add origin {0}'.format(fabric_settings.get('git_repo_url')))
-            local('git checkout -b develop')
+    current_path = os.getcwd()
+    os.chdir("./{}".format(project_name))
+    run('git init', warn=True)
+    run('git remote add origin {0}'.format(fabric_settings.get('git_repo_url')), warn=True)
```

Comment:
@rmutter What's nice about the config being in memory is that I just have to add a method to validate the fields! TODO to be added in `yurt/yurt_core::new_project`